.github/workflows/e2e-beta-tests.yml

```
@@ -0,0 +1,568 @@
+            fi
+            docker save "$HA_IMAGE_GHCR" -o /tmp/ha-image.tar
+            echo "Pulled and cached from $registry"
+            exit 0
+          fi
+          echo "Failed to pull from $registry, trying next..."
+          sleep 15
+        done
+        echo "All registries failed" && exit 1
+
+    - name: Run full E2E test suite
+      run: |
```

actionlint (1.7.12)

[error] shellcheck reported issue in this script: SC2086:info:3:15: Double quote to prevent globbing and word splitting

(shellcheck)